ksqldb-execution/src/main/java/io/confluent/ksql/execution/util/ComparisonUtil.java

```diff
@@ -15,38 +15,104 @@
 
 package io.confluent.ksql.execution.util;
 
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableList;
 import io.confluent.ksql.execution.expression.tree.ComparisonExpression;
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression.Type;
 import io.confluent.ksql.schema.ksql.SqlBaseType;
 import io.confluent.ksql.schema.ksql.types.SqlType;
 import io.confluent.ksql.util.KsqlException;
+import java.util.List;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
 
 final class ComparisonUtil {
 
-  private ComparisonUtil() {
+  private static final List<Handler> HANDLERS = ImmutableList.<Handler>builder()
+      .add(handler(SqlBaseType::isNumber, ComparisonUtil::handleNumber))
+      .add(handler(SqlBaseType.STRING, ComparisonUtil::handleString))
+      .add(handler(SqlBaseType.BOOLEAN, ComparisonUtil::handleBoolean))
+      .build();
 
+  private ComparisonUtil() {
   }
 
-  static boolean isValidComparison(
+  static void isValidComparison(
```

Comment:
Found a bug in here when moving tests to QTT:  Comparisons to the `NULL` type result in `SqlType` of `null`, which caused an NPE in this method!

Unfortunately, fixing it caused the complexity of this method to go over the checkstyle rule, so had to refactor.